### PR TITLE
407 - Docket Section Work Queue Endpoints

### PR DESCRIPTION
**Pull request by codyseibert**
- adding a /v1/workitems?section=docket endpoint for getting the workitems associated with the docket section
- changing the api logic to no longer hard code the work item to the "docketclerk" user after upload, but instead put it in the docket section
- adding a default message for when a document is uploaded in the work item

### CODE REVIEW

File: shared/src/business/useCases/fileGenericDocument.interactor.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  const user = new User({ userId });
+
+  const documentId = await applicationContext
+    .getPersistenceGateway()
+    .uploadDocument({
+      applicationContext,
+      document,
+    });
+
+  const caseEntity = getCaseEntityForUpload({
```
Comment (dknesek) [inline shared/src/business/useCases/fileGenericDocument.interactor.js:33]: This belongs in the `associateRespondentDocumentToCase` interactor. Remember that with an upload interactor, only the part up through the persistence call runs in the client. The remainder is delegated to the second interactor (that runs on the server).

File: shared/src/business/useCases/respondent/fileRespondentDocument.interactor.js [hunk 2]
```diff
@@ -26,20 +29,16 @@ exports.fileAnswer = async ({
       document,
     });
 
+  const caseEntity = getCaseEntityForUpload({
```
Comment (dknesek) [inline shared/src/business/useCases/respondent/fileRespondentDocument.interactor.js]: This belongs in `associateRespondentDocumentToCase`.

**Review by dknesek — DISMISSED** [review]
Nice work, Cody.

File: shared/src/business/useCases/fileGenericDocument.interactor.js (hunk 1 above)
Comment (codyseibert) [reply shared/src/business/useCases/fileGenericDocument.interactor.js:33]: Ok, so the issue I'm having is we need to know what to put in "caseToUpdate" when we invoke associateRespondentDocumentToCase which involves a lot of knowledge of how to build up the case object.  Maybe our useCase shouldn't need to know so much?

Comment (codyseibert) [reply shared/src/business/useCases/fileGenericDocument.interactor.js:33]: For example, we need to know that we have to attach a document to the case.documents array, and that we need to attach a workitem to the document, and that we need to attach a message to the work item.

**Review by ericsorenson — APPROVED** [review]

**Pull request merged**